Xamarin.Forms.Core/BindableObjectExtensions.cs

```diff
@@ -1,10 +1,37 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.Linq.Expressions;
 
 namespace Xamarin.Forms
 {
 	public static class BindableObjectExtensions
 	{
+		internal static void PropagateBindingContext<T>(this BindableObject self, IList<T> children)
+		{
+			PropagateBindingContext(self, children, BindableObject.SetInheritedBindingContext);
+		}
+
+		internal static void PropagateBindingContext<T>(this BindableObject self, IList<T> children, Action<BindableObject, object> setChildBindingContext)
+		{
+			var gotBindingContext = false;
+			object bc = null;
+
+			for (var i = 0; i < children.Count; i++)
+			{
+				var bo = children[i] as BindableObject;
+				if (bo == null)
+					continue;
+
+				if (!gotBindingContext)
+				{
+					bc = self.BindingContext;
+					gotBindingContext = true;
+				}
+
+				setChildBindingContext(bo, bc);
+			}
```

Comment:
can we rewrite this monstrosity (not yours) to get rid of the `gotBindingContext` ?

something like

```csharp
{
    if (children == null || children.Count == 0)
        return;
    var bc = self.BindingContext;
    for (var i = 0; i < children.Count; i++)
    {
        var bo = children[i] as BindableObject;
        if (bo == null)
           continue;
        setChildBindingContext(bo, bc);
 }